mysqlnd_azure.c

```diff
@@ -277,111 +316,143 @@ MYSQLND_METHOD(mysqlnd_azure_data, connect)(MYSQLND_CONN_DATA ** pconn,
 	{
 		SET_CONNECTION_STATE(&conn->state, CONN_READY); //set ready status so the connection can be closed correctly later if redirect succeeds
 
+		DBG_ENTER("[redirect] mysqlnd_azure_data::connect::redirect");
+        char redirect_host[MAX_REDIRECT_HOST_LEN] = { 0 };
+		char redirect_user[MAX_REDIRECT_USER_LEN] = { 0 };
+		unsigned int ui_redirect_port = 0;
+		zend_bool serverSupportRedirect =  get_redirect_info(conn, redirect_host, redirect_user, &ui_redirect_port);
+		if(!serverSupportRedirect) {
+            if(MYSQLND_AZURE_G(enableRedirect) == REDIRECT_ON) {
+                //When REDIRECT_ON, if there is no redirection information contained in the last_message, then redirection is not possible. In this case, abort the connection
+                conn->m->send_close(conn);
+                SET_CLIENT_ERROR(conn->error_info, -1, UNKNOWN_SQLSTATE, "Connection aborted because redirection is not enabled on the MySQL server or the network package doesn't meet meet redirection protocol.");
```

Comment:
Perhaps it's better to use a constant macro instead of the hard-coded `-1` here, and elsewhere in this file?